### ISSUE TITLE
[New Component] Swatch Group + Swatch

### ISSUE THREAD

**Issue by macandcheese** [Esri/calcite-design-system#10288]
### Description

Create a new Swatch Group + Swatch component that provides selection affordances, keyboard navigation, and other accessible helpers for the Swatch Group / Swatch.

### User Stories

As a designer, I want to create a selection experience for Swatches. I may want to have multiple swatch groups, and they may exist within any number of workflow locations and contexts - not just within a Color Picker component's "Saved" area.

It should not be scoped to "Color" workflows - as it may display patterns or other images.

### Acceptance Criteria

A public Swatch Group and Swatch component set is available. Calcite Color Picker Swatch is deprecated.

### Relevant Info

[Initial Figma design spec](https://www.figma.com/design/rneB6YLEuJRKazHnk0vVzv/Swatch-Group-%2B-Swatch?node-id=301-305&m=dev&focus-id=1631-15271) includes the following:

- Revised Swatch active/selected state
- New Swatch disabled state
- New Swatch image-fill option
- Revised Swatch hover and focus styling
- Swatch Group spacing copied from existing saved swatch spec from Color Picker
- Swatch Groups should support our typical array of `selection-modes`
- [Clickable prototype](https://www.figma.com/proto/rneB6YLEuJRKazHnk0vVzv/Swatch-Group-%2B-Swatch?page-id=301%3A305&node-id=1620-12945&viewport=104%2C290%2C0.87&t=3d7lPlFSG4YieAX4-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=1620%3A12945&show-proto-sidebar=1) with selection, hover, and focus in light and dark modes (Color Picker used for rough context only).

Final design may adjust based on feedback and usability testing during development.

### Helpful Details

Blocked issues: #748, #1616 

### Priority impact

impact - p2 - want for an upcoming milestone

### Esri team

Calcite (design)

**monday.com sync:** #9367014964

**Comment by ashetland** [comment]
Added design info to Relevant Info section above. Final design may be updated during development phase based on usability discoveries and feedback.

**Comment by macandcheese** [comment]
Yeah, could you log a follow up to resolve that? Might need to adjust the internally rendered svg shape. I'd consider that a low priority /  3 estimate follow up.

**Comment by DitwanP** [comment]
🍠✨ Verified by @isaacbraun and myself on `3.3.0-next.56` and locally.

https://codepen.io/isaacbraun/pen/raOraKz

Follow up issue created to resolve the incorrect border radius behavior: https://github.com/Esri/calcite-design-system/issues/12762
